### PR TITLE
UPD Ignore stop command if location does not exist

### PR DESCRIPTION
**Pull request by michalxo**

### CODE REVIEW

**Comment by michalxo** [comment]
This should be the new PR version. Ready to be merged to master. Waiting on @orpiske to confirm if this works for you as well. 
Thank you

File: README.md
Comment (orpiske) [inline README.md]: The variable needs to be named like amq_broker_external_address in order to prevent confusion with similar variables used in other roles.

File: tasks/get_amq_broker_distribution.yml
Comment (orpiske) [inline tasks/get_amq_broker_distribution.yml:78]: I think it would be good to be able to turn on/off the certificate validation (via validate_certs option).

File: tasks/service_stop_amq_broker_instance.yml
Comment (orpiske) [inline tasks/service_stop_amq_broker_instance.yml]: A better error message must be provided here. "Bad artemis=<pid>" does not provide any clue about what happen. Please provide a message that explains what happened and what should be done to resolve the issue.

File: tasks/service_stop_amq_broker_instance.yml
Comment (orpiske) [inline tasks/service_stop_amq_broker_instance.yml]: **NOT AN ISSUE**: I am wondering if we should include procps-ng as a dependency that is checked. Thoughts?

File: defaults/main.yml
Comment (orpiske) [inline defaults/main.yml]: I think we should default to /var instead of /tmp. Data on tmp is transient. It is removed on reboot ... On var is it not. There may be situations where we might need to reboot the broker host, but we still want to keep the facts file between reboots.

File: tasks/service_stop_amq_broker_instance.yml
Comment (orpiske) [reply tasks/service_stop_amq_broker_instance.yml]: Also, I'd recommend renaming bad_artemis_pid to something that properly represent its intent/content (ie.: broker_stuck_pid, etc).

File: tasks/service_stop_amq_broker_instance.yml
Comment (orpiske) [reply tasks/service_stop_amq_broker_instance.yml]: I think this is very dangerous. It can kill broker instances not running on the same user (this is a problem for Maestro). 

1. I don't think it needs to become root here.
2. I think a safer command line, using shell module instead of command, would be something like: 

> pgrep -U $(id -u {{ amq_broker_user }} ) -f '.\*java.\*artemis.\*'

**Review by orpiske — CHANGES_REQUESTED** [review]
All in all, pretty good. There's only a few minor things to change before merging.

File: README.md
Comment (michalxo) [reply README.md]: This is not related to AMQ broker at all, but to node, so adding amq_broker prefix does not make sense to me.
Unless you want to stress, that it is applicable for this role only.

Specially this is for Openstack VMs, which operate on 172 network, while outside access is through floating IP, which is completely different network - which even node itself does not know about..

File: defaults/main.yml
Comment (michalxo) [reply defaults/main.yml]: Here I would like to differ. The facts are not only "static ones", but are dependent on current profile/deployment.
For example, facts will contain also HA/Clustered status (master/slave, extra params), shared store, etc.

So these are not only ansible_facts, but mostly facts related to dtests ExteranlBroker, to know, what configuration is deployed, so we can operate it properly.

Please see templates/amq_broker_facts_template.j2 for details. Thus I would like to keep it in tmp.

File: tasks/get_amq_broker_distribution.yml
Comment (michalxo) [reply tasks/get_amq_broker_distribution.yml:78]: You mean when downloading packages from url as general for whole role?

File: tasks/service_stop_amq_broker_instance.yml
Comment (michalxo) [reply tasks/service_stop_amq_broker_instance.yml]: Agree, I will update it.

File: tasks/service_stop_amq_broker_instance.yml
Comment (michalxo) [reply tasks/service_stop_amq_broker_instance.yml]: Nope that won't work in some cases. For example if broker was started by root, or any other user, who is *not* current {{ amq_broker_user }}, it will always FAIL and whole deployment would be wrong. Thus I added there strong kill for all artemis processes. 
As another option I was thinking about adding another parts of artemis CMD for example "-Dartemis.home" or similar to distinguish clearly brokers.

Also, your pgrep ... command is same as pkill -u {{ amq_broker_user }} ... - so it can be simplified a lot.

File: README.md
Comment (orpiske) [reply README.md]: Yes, I do want to stress that. Thanks.

File: tasks/get_amq_broker_distribution.yml
Comment (orpiske) [reply tasks/get_amq_broker_distribution.yml:78]: When downloading the db driver.

File: tasks/service_stop_amq_broker_instance.yml
Comment (orpiske) [reply tasks/service_stop_amq_broker_instance.yml]: Yes, I meant pkill, not pgrep. That said, it should not kill Artemis processes not belonging to amq_broker_user.

File: README.md
Comment (orpiske) [reply README.md]: Yes, I do want to stress that it is related to the amq_broker role. That's how all user variables used by this roles should be named.

File: tasks/service_stop_amq_broker_instance.yml
Comment (michalxo) [reply tasks/service_stop_amq_broker_instance.yml]: <mtoth> what if we would put there amq_broker_allow_kill_all bool?
<opiske> if it defaults to false, then yes

File: README.md
Comment (michalxo) [reply README.md]: Resolved. amq_broker_ prefix added.

File: tasks/service_stop_amq_broker_instance.yml
Comment (michalxo) [reply tasks/service_stop_amq_broker_instance.yml]: Commited changes to reflect this behavior. Please see latest commit(s).

File: tasks/get_amq_broker_distribution.yml
Comment (michalxo) [reply tasks/get_amq_broker_distribution.yml:78]: Implemented in the last commit. Easyfix.

**Comment by michalxo** [comment]
Please see my latest changes for fixes.
Hopefully this is ready for merge.
Thank you for review.

**Review by orpiske — COMMENTED** [review]
I don't see the validate_certs option, as documented here, in the code. Are you sure you added it? https://docs.ansible.com/ansible/latest/modules/get_url_module.html

Yes you did add it. Github UI confused me.

**Review by orpiske — APPROVED** [review]

**Comment by orpiske** [comment]
LGTM after the changes.

**Comment by michalxo** [comment]
Validate certs commit was added a few minutes later than original PR. 
Thanks! Hopefully all will work. Please verify with me tomorrow. If help needed, I can fix it asap.

**Pull request merged**
